OrbitQt/servicedeploymanager.cpp

```diff
@@ -313,13 +375,14 @@ outcome::result<void> ServiceDeployManager::StartOrbitServicePrivileged() {
 
   orbit_service_task_->Start();
 
-  OUTCOME_TRY(loop_.exec());
+  OUTCOME_TRY(loop_->exec());
   QObject::connect(&orbit_service_task_.value(), &OrbitSshQt::Task::errorOccurred, this,
                    &ServiceDeployManager::handleSocketError);
   return outcome::success();
 }
 
 outcome::result<void> ServiceDeployManager::InstallOrbitServicePackage() {
+  CHECK(QThread::currentThread() == thread());
```

Comment:
How does this work? From Qt: currentThread "Returns a pointer to a QThread which manages the currently executing thread." But what is thread() returning?

Reply:
`thread()` is `this->thread()`, so it calls `QObject::thread` [details](https://doc.qt.io/qt-5/qobject.html#thread)

Every QObject has an associated thread (which can be read by QThread::thread and changed by QObject::moveToThread). Whenever this object receives an event, the event dispatcher will determine the associated thread and queue an event handler task (a slot execution for example) on this thread's event loop.

Reply:
Ok, thanks for the info!